component/http/v2/check.go

```diff
@@ -0,0 +1,65 @@
+package v2
+
+import (
+	"net/http"
+)
+
+// AliveStatus type representing the liveness of the service via HTTP component.
+type AliveStatus int
+
+// ReadyStatus type.
+type ReadyStatus int
+
+const (
+	// Alive represents a state defining a Alive state.
+	Alive AliveStatus = 1
+	// Unresponsive represents a state defining a Unresponsive state.
+	Unresponsive AliveStatus = 2
+
+	alivePath = "/alive"
+
+	// Ready represents a state defining a Ready state.
+	Ready ReadyStatus = 1
+	// NotReady represents a state defining a NotReady state.
+	NotReady ReadyStatus = 2
+
+	readyPath = "/ready"
+)
+
+// ReadyCheckFunc defines a function type for implementing a readiness check.
+type ReadyCheckFunc func() ReadyStatus
+
+// LivenessCheckFunc defines a function type for implementing a liveness check.
+type LivenessCheckFunc func() AliveStatus
+
+// LivenessCheckRoute returns a route for liveness checks.
+func LivenessCheckRoute(acf LivenessCheckFunc) *Route {
+	f := func(w http.ResponseWriter, r *http.Request) {
+		switch acf() {
+		case Alive:
+			w.WriteHeader(http.StatusOK)
+		case Unresponsive:
+			w.WriteHeader(http.StatusServiceUnavailable)
+		default:
+			w.WriteHeader(http.StatusOK)
+		}
+	}
+	route, _ := NewRoute(http.MethodGet, alivePath, f)
+	return route
+}
+
+// ReadyCheckRoute returns a route for ready checks.
+func ReadyCheckRoute(rcf ReadyCheckFunc) *Route {
+	f := func(w http.ResponseWriter, r *http.Request) {
+		switch rcf() {
+		case Ready:
+			w.WriteHeader(http.StatusOK)
+		case NotReady:
+			w.WriteHeader(http.StatusServiceUnavailable)
+		default:
+			w.WriteHeader(http.StatusOK)
```

Comment:
@gmaz42 
I agree. The same goes for the liveness. 
Question:  Should the default be the opposite, a failure? Just to be on the safe side?

Reply:
Good question; first let's try to define what a hit on `default:` means here :thinking: As far as I understand this happens if the ReadyCheck function returns something invalid. So perhaps it is a good idea to make it fail for all cases which are not `Ready`? And when it is not `NotReady` it could additionally log something like "check what you are doing, you returned an invalid enum from the ReadyCheck function" (thanks to Go's lack of enum support.. :smiling_face_with_tear: )

Reply:
Done.